examples/xgboost/custom/xgboost_bagging_aggregator.py

```diff
@@ -0,0 +1,183 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import shutil
+
+from nvflare.apis.dxo import DXO, DataKind, from_shareable
+from nvflare.apis.fl_constant import FLContextKey, ReservedKey, ReturnCode
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.abstract.aggregator import Aggregator
+from nvflare.app_common.app_constant import AppConstants
+
+
+class XGBoostBaggingAggregator(Aggregator):
+    def __init__(
+        self,
+        expected_data_kind: DataKind = DataKind.WEIGHTS,
+    ):
+        """Perform bagging aggregation for XGBoost trees.
+
+        The trees are pre-weighted during training.
+        Bagging aggregation simply add the new trees to existing global model.
+
+        Args:
+            expected_data_kind (Union[DataKind, Dict[str, DataKind]]):
+                DataKind for DXO. Defaults to DataKind.WEIGHTS
+                Indicating the tree representation given by XGBoost
+        """
+        super().__init__()
+        self.logger.debug(f"expected data kind: {expected_data_kind}")
+        self.history = []
+        self.expected_data_kind = expected_data_kind
+        # Check expected data kind
+        if expected_data_kind != DataKind.WEIGHTS:
+            raise ValueError(f"expected_data_kind = {expected_data_kind} is not {DataKind.WEIGHTS}")
+
+    def accept(self, shareable: Shareable, fl_ctx: FLContext) -> bool:
+        """Store shareable and update aggregator's internal state
+
+        Args:
+            shareable: information from contributor
+            fl_ctx: context provided by workflow
+
+        Returns:
+            The first boolean indicates if this shareable is accepted.
+            The second boolean indicates if aggregate can be called.
+        """
+        try:
+            dxo = from_shareable(shareable)
+        except BaseException:
+            self.log_exception(fl_ctx, "shareable data is not a valid DXO")
+            return False
+
+        contributor_name = shareable.get_peer_prop(key=ReservedKey.IDENTITY_NAME, default="?")
+        contribution_round = shareable.get_header(AppConstants.CONTRIBUTION_ROUND)
+
+        rc = shareable.get_return_code()
+        if rc and rc != ReturnCode.OK:
+            self.log_warning(fl_ctx, f"Contributor {contributor_name} returned rc: {rc}. Disregarding contribution.")
+            return False
+
+        # Accept expected DXO in shareable
+        if not isinstance(dxo, DXO):
+            self.log_error(fl_ctx, f"Expected DXO but got {type(dxo)}")
+            return False
```

Comment:
this check is not needed as you already have 

```
        try:
            dxo = from_shareable(shareable)
```

Reply:
removed